### PR TITLE
Add functions for mounting and unmounting filesystems

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Added functions:
 - bd_fs_mount
 - bd_fs_unmount

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

File: src/plugins/fs.c [hunk 1]
```diff
@@ -291,6 +308,481 @@ static gint synced_close (gint fd) {
...
+            wpid = waitpid (pid, &status, WUNTRACED | WCONTINUED);
+            if (wpid == -1) {
+                g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_FAIL,
+                             "Error while waiting for process.");
+                return FALSE;
+            }
+
+            if (WIFEXITED (status)) {
+              if (WEXITSTATUS (status) != EXIT_SUCCESS) {
+                  channel = g_io_channel_unix_new (pipefd[0]);
```
Comment (vpodzime) [inline src/plugins/fs.c]: The channel needs to be shut down and unrefed when no longer used.

File: src/plugins/fs.c [hunk 2]
```diff
@@ -291,6 +308,481 @@ static gint synced_close (gint fd) {
...
+                             "Unsupported argument for unmount: '%s'", (*extra_p)->opt);
+                return FALSE;
+            }
+        }
+    }
+
+    if (run_as_uid != current_uid || run_as_gid != current_gid) {
+        return run_as_user (&do_unmount, &args, run_as_uid, run_as_gid, error);
+    } else
+       return do_unmount (&args, error);
```
Comment (vpodzime) [inline src/plugins/fs.c]: bad indentation here.

File: src/lib/plugin_apis/fs.api [hunk 3]
```diff
@@ -204,6 +205,36 @@ GType bd_fs_vfat_info_get_type () {
...
 gboolean bd_fs_wipe (const gchar *device, gboolean all, GError **error);
 
+/**
+ * bd_fs_unmount:
+ * @spec: mount point or device to unmount
+ * @lazy: enable/disable lazy unmount
+ * @force: enable/disable force unmount
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the unmount
+ *                                                 currently only 'run_as_uid'
+ *                                                 and 'run_as_gid' is supported
```
Comment (vpodzime) [inline src/lib/plugin_apis/fs.api]: Please also document how the value (``key``) should look like (and that it cannot be ``"0"``).

**Pull request merged**